numpy/lib/__init__.py

```diff
@@ -74,13 +66,23 @@ def __getattr__(attr):
     import math
     import warnings
 
+    def _private_module_msg(module_name: str) -> str:
+        return (
+            f"`np.lib.{module_name}` is now private. Function that you're "
+            "looking for should be available from the main namespace `np.*`"
+        )
```

Comment:
The wording here needs some adjustment. It's possible someone is using a private function that was removed on purpose. Maybe "If you are using a public function, it should be available in the main numpy namespace, otherwise check the NumPy 2.0 migration guide."

Reply:
Sure! I updated error message here.

Reply:
Thanks! I don't see `arraysetops` there, which is currently addressed in astropy/astropy#15253 but found I can also access its attributes as
```python
import numpy.lib.arraysetops
numpy.lib.arraysetops.__all__
```
– would that be a viable solution (ultimately we may not need that list at all if the functions are no longer visible)?

Reply:
All functions provided by `arraysetops` are available through the main namespace `np.*`, therefore `arraysetops` was made private, and shouldn't be accessed. Here's an issue tracking `lib` submodules changes: https://github.com/numpy/numpy/issues/24507

Reply:
The warning in `__getattr__` in `numpy/lib/__init__.py` is taking care of it for `lib` namespace.

Reply:
If it's for [this line](https://github.com/astropy/astropy/blob/7bc600506dad3c1ce697a68fcbbc395256a438a4/astropy/utils/masked/function_helpers.py#L136)I think the correct thing is just to explicitly list the names from the main namespace that you want to ignore.

Reply:
Yes, that's what is done right now; I was merely worried that list might change at a later point, but then it's probably to safe to just use those, as listed in #24507 as well.